### PR TITLE
[Enhancement](filter) support only min/max runtime filter in BE

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
this PR https://github.com/apache/doris/pull/25193 have achieve about FE.
eg: select count() from lineorder join supplier on lo_partkey < s_suppkey;
will have a max filter after build hash table , so could use it to filter probe table data.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/exprs/minmax_predicate.h [hunk 1]
```diff
@@ -128,11 +128,158 @@ class MinMaxNumFunc : public MinMaxFuncBase {
...
+            if (other_minmax->_max > this->_max) {
+                this->_max = other_minmax->_max;
+            }
+        }
+
+        return Status::OK();
+    }
+
+    //max filter the min is useless, so return nullptr directly
+    void* get_min() override { return nullptr; }
```
Comment (HappenLee) [inline be/src/exprs/minmax_predicate.h:277]: better DCHECK(false) core here, not return nullptr

File: be/src/exprs/minmax_predicate.h [hunk 2]
```diff
@@ -128,11 +128,158 @@ class MinMaxNumFunc : public MinMaxFuncBase {
...
+            return false;
+        }
+
+        T val_data = *reinterpret_cast<T*>(data);
+        return val_data <= this->_max;
+    }
+
+    Status merge(MinMaxFuncBase* minmax_func, ObjectPool* pool) override {
+        if constexpr (std::is_same_v<T, StringRef>) {
+            MinMaxNumFunc<T>* other_minmax = static_cast<MinMaxNumFunc<T>*>(minmax_func);
```
Comment (HappenLee) [inline be/src/exprs/minmax_predicate.h:258]: assert_cast to  MaxNumFunc<T>*

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**